railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -115,6 +117,19 @@ def extract_environment_from_path(path)
           available_environments.find { |env| path.include? env } if path.end_with?(".yml.enc")
         end
 
+        # Check for a mismatch between Rails::Command.environment and option set by --environment flag.
+        # This warning should only occur in non-development environments to avoid confusing the user
+        # when trying to perform this action against different environment modes.
+        def warn_if_environment_mismatch(method_name: :show)
+          if Rails::Command.environment != "development" && Rails::Command.environment != options[:environment]
```

Comment:
I think we generally try to avoid calling environments inside Rails and try to use a configuration instead. See: https://github.com/rails/rails/pull/42828#issuecomment-884470990

Reply:
@p8 Sorry to be bothering with simple questions. So I went through the mentioned PR and understood that we can use values from `request.env` in certain places. In our case, it is a command line execution right? So, ideally that can't be possible from request object.

So should I be looking in for a different variable that holds the configuration?

Reply:
No problem 🙂 You can ignore the request.env stuff. I think we should add a config option, maybe config.credentials.allow_editing, that is only set to true for the development environment. 
Similar to: https://github.com/rails/rails/blob/52db7f2ef3c4a902227cac163657c89282986e25/railties/lib/rails/application/configuration.rb#L37 and https://github.com/rails/rails/blob/52db7f2ef3c4a902227cac163657c89282986e25/railties/lib/rails/application/configuration.rb#L72